python/lsst/ts/electrometer/electrometerCSC/ElectrometerCSC.py

```diff
@@ -1,480 +1,734 @@
+        self.mainConfiguration = FileReaderYaml("./settingFiles", "", "")
+        self.mainConfiguration.loadFile("mainSetup")
+
+        # Pre-SAL setup
+        self.fitsDirectory = os.path.join(
+            str(self.mainConfiguration.readValue('filePath')), f"{index}")
+        if not os.path.exists(self.fitsDirectory):
+            os.makedirs(self.fitsDirectory)
+
+        self.salId = self.mainConfiguration.readValue('salId')
+        self.simulated = self.mainConfiguration.readValue('simulated')
+
+        if initial_state not in base_csc.State:
+            raise ValueError(
+                f"intial_state={initial_state} is not a salobj.State enum")
+        super().__init__(SALPY_Electrometer, self.salId)
+
+        # CSC declarations
+        self.summary_state = initial_state
+
+        # Loops
+        self.stateCheckLoopfrequency = 0.2
+        self.telemetryLoop = 0.2
+
+        # Loggins start
+        self.log = logging.getLogger(__name__)
+        self.log.debug("logger initialized")
+        self.log.info("Electrometer CSC initialized")
+
+        # Events declaration
+        self.evt_appliedSettingsMatchStart_data = self.evt_appliedSettingsMatchStart.DataType()
+        self.evt_detailedState_data = self.evt_detailedState.DataType()
+        self.evt_digitalFilterChange_data = self.evt_digitalFilterChange.DataType()
+        self.evt_errorCode_data = self.evt_errorCode.DataType()
+        self.evt_integrationTime_data = self.evt_integrationTime.DataType()
+        self.evt_intensity_data = self.evt_intensity.DataType()
+        self.evt_largeFileObjectAvailable_data = self.evt_largeFileObjectAvailable.DataType()
+        self.evt_measureRange_data = self.evt_measureRange.DataType()
+        self.evt_measureType_data = self.evt_measureType.DataType()
+        self.evt_settingsAppliedReadSets_data = self.evt_settingsAppliedReadSets.DataType()
+        self.evt_settingsAppliedSerConf_data = self.evt_settingsAppliedSerConf.DataType()
+        self.evt_settingVersions_data = self.evt_settingVersions.DataType()
+
+        # Electrometer declarations
+        if(self.simulated == 1):
+            self.electrometer = ElectrometerSimulator()
+        else:
+            self.electrometer = ElectrometerController()
+        self.appliedSettingsMatchStart = False
+        self.detailed_state = iec.ElectrometerStates.OFFLINESTATE
+        self.stop_triggered = False
+        self.lastScanTime = 0
+
+        # Loop initialization
+        asyncio.ensure_future(self.init_stateLoop())
+        asyncio.ensure_future(self.init_intensityLoop())
+
+        # salobj configuration to run the same command mult
+        self.cmd_startScanDt.allow_multiple_commands = True
+        self.cmd_startScan.allow_multiple_commands = True
+        self.cmd_stopScan.allow_multiple_commands = True
+
+    def do_setLogLevel(self):
+        pass
+
+    def do_setSimulationMode(self):
+        pass
+
+    def do_start(self, id_data):

```

Comment:
These overrides would be cleaner using the provided hooks of either `begin_transition` or `end_transition`.